### ISSUE TITLE
pip install mediapipe not working on apple silicon Macs, even in anaconda environment

### ISSUE THREAD

**Issue by nakshatrasinghh** [google/mediapipe#2120]
<img width="787" alt="Screenshot 2021-06-03 at 7 39 17 PM" src="https://user-images.githubusercontent.com/53419293/120658831-69e10500-c4a3-11eb-9b41-aed10ef79007.png">

**Comment by kiranscaria** [comment]
What is the pip version?

**Comment by nmk987** [comment]
Same issue.
Pip version: 21.1.2 
Python: 3.8.6

**Comment by nakshatrasinghh** [comment]
> Same issue.
> Pip version: 21.1.2
> Python: 3.8.6

Same versions on my machine as well. Any ways to solve this issue?

**Comment by sgowroji** [comment]
Hi @nakshatrasinghh , Can you check this [reference](MediaPipe PyPI supports macOS 10.15 and above. See https://github.com/google/mediapipe/blob/master/docs/getting_started/troubleshooting.md#python-pip-install-failure.).

**Comment by nakshatrasinghh** [comment]
> Did you try it with python 3.7? I myself got some issues on my macbook (11.2.2). After I changed the conda python env to 3.7 everything worked fine

Did you try conda install MediaPipe or pip3 install MediaPipe?

**Comment by jiuqiant** [comment]
MediaPipe currently doesn't support arm macOS. So does Mediapipe Python package.

**Comment by josiahsrc** [comment]
I did some digging into this and went so far as to attempt to build a wheel for arm64, but to no avail. The only workaround I could find was to instead use rosetta. Here are the steps I took, which may be of help to others facing m1 issues:

1. Write your terminal application from within finder, select get info, and enable `open using rosetta'
2. Open up your terminal application with the rosetta flag set
3. Create a virtual env in your project, `python3 -m venv myvenv`
4. Activate it, `source ./myvenv/bin/activate`
5. `pip install mediapipe`

After this, I was able to run media pipe with my Mac. I hope this helps, and good luck!

**Comment by sgowroji** [comment]
Hi @josiahsrc, Thank you so much for sharing your answer this will definitely help full to the community. Thanks!

**Comment by belelaritra** [comment]
> I did some digging into this and went so far as to attempt to build a wheel for arm64, but to no avail. The only workaround I could find was to instead use rosetta. Here are the steps I took, which may be of help to others facing m1 issues:
> 
> 1. Write your terminal application from within finder, select get info, and enable `open using rosetta'
> 2. Open up your terminal application with the rosetta flag set
> 3. Create a virtual env in your project, `python3 -m venv myvenv`
> 4. Activate it, `source ./myvenv/bin/activate`
> 5. `pip install --upgrade pip`
> 6. `pip install mediapipe`
> 
> After this, I was able to run mediapipe on my Mac. I hope this helps, and good luck!

Yes after this process it installed successfully into my machine but it's not working. Showing wrong architecture.

**Comment by josiahsrc** [comment]
@belelaritra Make sure you're using a python3.8.* version installed with x86_64 architecture. I was able to do this by installing Homebrew for x86_64 architecture (see [this](https://stackoverflow.com/questions/64963370/error-cannot-install-in-homebrew-on-arm-processor-in-intel-default-prefix-usr) post). Then, I installed python 3.8.* using `arch -x86_64 brew install python@3.8`. From there, I ran the steps in my post above.

**Comment by belelaritra** [comment]
Ohh !! Sorry, I'm using arm64 architecture mac (apple silicon).

**Comment by belelaritra** [comment]
here homebrew is installed inside `opt` folder.

**Comment by josiahsrc** [comment]
On m1 macs, its typical to have two versions of Homebrew installed. One will be in opt folder (Homebrew for ARM), the other in /usr/local (Homebrew for x86_64)

**Comment by belelaritra** [comment]
I have followed all the steps from the [link](https://stackoverflow.com/questions/64963370/error-cannot-install-in-homebrew-on-arm-processor-in-intel-default-prefix-usr) you have provided but there is no `Celler` folder inside `homebrew` but it's inside `local` only.

**Comment by josiahsrc** [comment]
Did you follow the [second post](https://stackoverflow.com/a/65398385/10880084)? It looks like the accepted one is a bit outdated. Make sure you use the x86_64 homebrew via `arch -x86_64 /usr/local/homebrew/bin/brew install python@3.8`. Apologies, I had forgotten that the ARM homebrew is usually default in PATH.

**Comment by belelaritra** [comment]
Sorry I followd the first one.
But then I followed your steps (just changed the command into `arch -x86_64 /usr/local/Cellar/python@3.8/3.8.11/bin/python3 -m venv myenv` ) and it's now working(just checked one hand module).

Thank you sir.
It helps me a lot.

**Comment by josiahsrc** [comment]
Happy to help! 😄

**Comment by Leon-Brant** [comment]
> MediaPipe 目前不支持 arm macOS。Mediapipe Python 包也是如此。
I see. Thank you 🙏 for sharing this information

**Comment by yasserdzx** [comment]
it doesnt work on python 3.10 i downgraded mine to 3.9 and it worked

**Comment by MatusGasparik** [comment]
Hi @jiuqiant. Can I ask what the plans are regarding the support for the **osx-arm64** platform?

**Comment by matanster** [comment]
Is the issue only with TensorFlow, or does anything else in mediapipe or which mediapipe depends play as a blocker with running mediapipe on the M1?

**Comment by iPHOENIX-matrix** [comment]
1. Create a new conda environment and install python version 3.9.13 in that environment
2. Then using pip install MediaPipe:
 Code:
 pip install mediapipe

Write this code in the terminal.
MediaPipe package will be installed.

I tried to install "mediapipe-silicon" in my M1 Pro MacBook but version error was coming every time with most versions of python but luckily I found the above solution and it worked. Hopefully this might work for other developers.

Happy Coding!
-iPHOENIX
